oxenss/rpc/client_rpc_endpoints.cpp

```diff
@@ -728,15 +753,147 @@ void batch::load_from(bt_dict_consumer params) {
             throw parse_error{"Invalid batch request: subrequests must have a method"};
         auto method = sr.consume_string_view();
         auto rpc_it = RequestHandler::client_rpc_endpoints.find(method);
-        if (rpc_it == RequestHandler::client_rpc_endpoints.end() || !rpc_it->second.load_subreq_bt)
+        if (rpc_it == RequestHandler::client_rpc_endpoints.end())
             throw parse_error{
                     "Invalid batch subrequest: invalid method \"" + std::string{method} + "\""};
         if (!sr.skip_until("params") || !sr.is_dict())
             throw parse_error{"Invalid batch request: subrequests must have a params dict"};
-        subreqs.push_back(rpc_it->second.load_subreq_bt(sr.consume_dict_consumer()));
+        subreqs.push_back(as_subrequest(rpc_it->second.load_req_bt(sr.consume_dict_consumer())));
     }
     if (subreqs.empty())
         throw parse_error{"Invalid batch request: empty \"requests\" list"};
 }
 
+// Copies an optional vector into a fixed-size array, substituting 0's for omitted vector elements,
+// and ignoring anything in the vector longer than the given size.  Gives nullopt if the input
+// vector is itself nullopt or empty.
+template <size_t N, typename T>
+static std::optional<std::array<T, N>> to_fixed_array(const std::optional<std::vector<T>>& in) {
+    if (!in || in->empty())
+        return std::nullopt;
+    std::array<T, N> out;
+    for (size_t i = 0; i < N; i++)
+        out[i] = i < in->size() ? (*in)[i] : T{0};
+    return out;
+}
+
+template <typename Dict>
+static void load_condition(ifelse& i, Dict if_) {
+    auto [height_ge_, height_lt_, hf_ge_, hf_lt_, v_ge_, v_lt_] = load_fields<
+            int,
+            int,
+            std::vector<int>,
+            std::vector<int>,
+            std::vector<int>,
+            std::vector<int>>(
+            if_,
+            "height_at_least",
+            "height_before",
+            "hf_at_least",
+            "hf_before",
+            "v_at_least",
+            "v_before");
+
+    auto hf_ge = to_fixed_array<2>(hf_ge_);
+    auto hf_lt = to_fixed_array<2>(hf_lt_);
+    auto v_ge = to_fixed_array<3>(v_ge_);
+    auto v_lt = to_fixed_array<3>(v_lt_);
+    auto height_ge = height_ge_;
+    auto height_lt = height_ge_;
+
+    if (!(height_ge_ || height_lt_ || hf_ge || hf_lt || v_ge || v_lt))
+        throw parse_error{"Invalid ifelse request: must specify at least one \"if\" condition"};
+
+    i.condition = [=](const snode::ServiceNode& snode) {
```

Comment:
this will capture `i`and `if_` by value, probably making a copy, very expensive and it does not use them in that lambda

Reply:
`[=]` will only copy variables that you use in the lambda